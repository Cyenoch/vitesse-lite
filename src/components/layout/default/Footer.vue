<template>
  <nav mt-6 inline-flex gap-2 text-xl>
    <button class="toggle-dark" @click="toggleDark()">
      <div i-carbon-sun dark:i-carbon-moon />
    </button>
  </nav>
</template>
