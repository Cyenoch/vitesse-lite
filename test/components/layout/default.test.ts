import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Footer from '~/components/layout/default/Footer.vue'

describe('component of Footer.vue', () => {
  it('should render', () => {
    const wrapper = mount(Footer, { props: {} })
    // expect(wrapper.text()).toContain('10')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be interactive', async () => {
    const wrapper = mount(Footer, { props: { } })

    toggleDark(false)

    expect(wrapper.find('.toggle-dark').exists()).toBe(true)

    await wrapper.get('.toggle-dark').trigger('click')

    expect(isDark.value).toBe(true)
  })
})
