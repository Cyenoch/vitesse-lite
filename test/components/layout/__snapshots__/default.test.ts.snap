// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component of Footer.vue > should render 1`] = `
"<nav mt-6="" inline-flex="" gap-2="" text-xl=""><button class="toggle-dark" icon-btn="">
    <div i-carbon-sun="" dark:i-carbon-moon=""></div>
  </button></nav>"
`;
